### PR TITLE
Photogram part 3: starting on UI

### PR DESCRIPTION
**Pull request by kcalva**

### CODE REVIEW

**Review by hcontr4 — APPROVED** [review]
Changes align with specs described in part 3.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,3 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
```
Comment (JoseMDO) [inline app/controllers/application_controller.rb:2]: Nice!

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -6,11 +6,23 @@
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
+    <%= render partial: "shared/cdn_assets" %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:9]: Great job adding your cdn assets to a partial to clean up the application layout file!

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -6,11 +6,23 @@
...
+
     <%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>
     <%= javascript_importmap_tags %>
   </head>
 
   <body>
-    <%= yield %>
+    <%= render partial: "shared/navbar" %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:16]: same here

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -6,11 +6,23 @@
...
+    <div class="container">
+      <% if notice.present? %>
+        <%= render partial: "shared/flash", locals: { message: notice, css_class: "success" } %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:19]: same here

File: app/views/shared/_navbar.html.erb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      <span class="navbar-toggler-icon"></span>
+    </button>
+    <div class="collapse navbar-collapse" id="navbarSupportedContent">
+      <ul class="navbar-nav mr-auto">
+        <li class="nav-item dropdown">
+          <a class="nav-link dropdown-toggle" href="#" role="button" data-bs-toggle="dropdown" aria-expanded="false">
+            Resources
+          </a>
+          <div class="dropdown-menu">
+            <a class="dropdown-item" href="/photos">Photos</a>
```
Comment (JoseMDO) [inline app/views/shared/_navbar.html.erb:15]: Remember we can start using the helper methods here like link_to :)

File: lib/tasks/dev.rake [hunk 6]
```diff
@@ -9,6 +9,20 @@ task({ :sample_data => :environment }) do
     User.destroy_all
   end
 
+  usernames = Array.new { Faker::Name.first_name }
+
+  usernames << "alice"
+  usernames << "bob"
+
+  usernames.each do |username|
```
Comment (JoseMDO) [inline lib/tasks/dev.rake:17]: Great job on your sample_data!

**Review by JoseMDO — APPROVED** [review]

**Review by ghost — COMMENTED** [review]
Wow, color me impressed! Great work Kevin!

**Pull request merged**